frontend/src/concepts/areas/types.ts

```diff
@@ -15,7 +16,9 @@ export type IsAreaAvailableStatus = {
   devFlags: { [key in string]?: 'on' | 'off' } | null; // simplified. `disableX` flags are weird to read
   featureFlags: { [key in FeatureFlag]?: 'on' | 'off' } | null; // simplified. `disableX` flags are weird to read
   reliantAreas: { [key in SupportedAreaType]?: boolean } | null; // only needs 1 to be true
-  requiredComponents: { [key in StackComponent]?: boolean } | null;
+  requiredComponents:
+    | { [key in DataScienceStackComponent]?: DataScienceClusterComponentStatus }
+    | null;
```

Comment:
Why change the API signature? We only care at this point whether the component is enabled or not.

Reply:
@christianvogt Fair point! I changed it to expose the full component status because the boolean approach was hiding useful information. With just true/false, you can't tell if a component is `"Removed"`, `"Unmanaged"`, or has version conflicts.

The richer object lets consumers (and developers debugging) actually see what's happening under the hood. Plus it gives us flexibility if we ever need that extra context in the UI.

But you're absolutely right that current consumers just need "available or not". I can easily keep the boolean API and do the `managementState !== 'Removed'` check internally. Want me to revert to the simpler approach?